pkg/hive/job/job.go

```diff
@@ -390,12 +410,19 @@ func (jt *jobTimer) start(ctx context.Context, wg *sync.WaitGroup, options optio
 
 		l.Debug("Timer job triggered")
 
+		stat.Start()
```

Comment:
We should do spans also for one-shot jobs. While they might be infinite loops they might also do some interesting one-shot initialization.

Reply:
I am worried about the carnality of one-shot jobs since they are also used to dynamically spawn goroutines like in https://github.com/cilium/cilium/blob/main/pkg/l2announcer/l2announcer.go#L702

Reply:
Wouldn't metrics be especially useful for those? Perhaps the solution is to not generate job names like that and use just "leader-election" as the name?

Reply:
So far, we only have jobs that live until shutdown, and these which live a very long time. In both cases, knowing how long they were running is not meaningful. It would only be meaningful to know for short lived jobs. Perhaps for one shots it makes sense to define the metric but only add jobs which have been marked with a `WithMetrics` option? That way we don't have unnecessary carnality

Reply:
Sounds good to me!